opennebula/helpers.go

```diff
@@ -92,3 +92,79 @@ func NoExists(err error) bool {
 
 	return false
 }
+
+// collect config IDs
+func collectListIDs(list []interface{}, attrName string) []int {
+
+	ids := make([]int, 0, len(list))
+
+	// collect IDs
+	for _, item := range list {
+		mapItem := item.(map[string]interface{})
+
+		id := mapItem[attrName].(int)
+
+		if id < 0 {
+			continue
+		}
+
+		ids = append(ids, id)
+	}
+
+	return ids
+}
+
+// return IDs that only appear on ref side
```

Comment:
It seems it is the opposite

Reply:
Code is right but this confirm that it's not really readable/maintainable.

In brief:
if `refID == id` then refID is in the second vector.
But I want to do a partial diff, i.e. the refID that isn't in the second vector

I'll try to update the code to be a bit more easy to understand.